rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -117,10 +102,16 @@ public TopologyHandler getHandler(TopologyDescriptor topologyDescriptor) {
             !PAResourceManagerProperties.RM_TOPOLOGY_ENABLED.getValueAsBoolean()) {
             throw new TopologyDisabledException("Topology is disabled");
         }
+        if (topologyDescriptor instanceof BestProximityDescriptor || topologyDescriptor instanceof ThresholdProximityDescriptor) {
```

Comment:
I don't like `instanceof`, especially in that case because adding a new TopologyDescriptor may require to update the condition and thus adding one more test. This could be avoided and simplified by adding a new method like `boolean requireDistanceComputing()` that returns `false` by default but is overriden on required implementations.